jsonnet/kube-prometheus/components/blackbox-exporter.libsonnet

```diff
@@ -152,6 +158,63 @@ function(params) {
     }],
   },
 
+  podSecurityPolicy:
+    local securityContext =
+      if bb.config.privileged then
+        {
+          privileged: true,
+          allowedCapabilities: ['NET_RAW'],
+          runAsUser: {
+            rule: 'RunAsAny',
+          },
+        }
+      else
+        {
+          privileged: false,
+          requiredDropCapabilities: ['ALL'],
+          runAsUser: {
+            rule: 'MustRunAsNonRoot',
+          },
+        };
```

Comment:
Hey, I'm having a hard time figuring out how to have this part inside an addon.

The `bb.config.privileged` is created at `blackbox-exporter.libsonnet` default values. Since this field is not created on main.libsonnet, I can't really pass it on to the addon... Or I just don't know how to do it.

Any suggestions on how to implement this blackbox-exporter bit as an addon?

Reply:
In `main.libsonnet` we have top level `blackboxExporter` object which has all fields  from `components/blackbox-exporter.libsonnet` so you can use `$.blackboxExporter.config.privileged` for detecting this.